### PR TITLE
feat: implement input module

### PR DESCRIPTION
**Pull request by jrandolf**

### CODE REVIEW

**Comment by sadym-chromium** [comment]
Please provide detailed description and e2e tests

**Comment by sadym-chromium** [comment]
Or convert it to draft, if it's not review-ready

File: src/bidiMapper/domains/context/browsingContextImpl.ts [hunk 1]
```diff
@@ -190,6 +190,14 @@ export class BrowsingContextImpl {
     return this.#parentId === null;
   }
 
+  get top(): BrowsingContextImpl {
```
Comment (thiagowfx) [inline src/bidiMapper/domains/context/browsingContextImpl.ts:193]: BrowsingContextStorage has a `findTopLevelContext` method already. Please reuse it.

Comment (jrandolf) [reply src/bidiMapper/domains/context/browsingContextImpl.ts:193]: Can there be more than one top-level context? In theory, it could be a forest, no?

Comment (jrandolf) [reply src/bidiMapper/domains/context/browsingContextImpl.ts:193]: Nevermind, my interpretation of the function was wrong.

Comment (jrandolf) [reply src/bidiMapper/domains/context/browsingContextImpl.ts:193]: Hmm, I think I'd rather keep this method and remove `findTopLevelContext`. `findTopLevelContext` is not used anywhere and it's far slower than this `BrowsingContextImpl.top`.

**Comment by thiagowfx** [comment]
In order to test against WPT, remove the `__dir__` files within `input`.

**Comment by thiagowfx** [comment]
Are you intending to break this PR down into smaller pieces? http://go/small-cls

**Comment by jrandolf** [comment]
@thiagowfx Depends. We could separate the schema and parser changes, but for the main implementation, do we want to have partial implementation?

**Comment by thiagowfx** [comment]
The splits that you did recently are a good start.

It's OK to keep the main implementation in a single PR as it's unnecessary burden for you to try to split them.

I was mostly referring to the helper/aux files and to the protocol/schema.

**Pull request closed without merging**